### ISSUE TITLE
AJAX Load Reinitialize Resets All Settings

### ISSUE THREAD

**Issue by newscloud** [kartik-v/yii2-widget-switchinput#23]
Hi Kartik, 

Great plugin. I'm trying to load new switches via AJAX.

values = $("input[name='place-chooser']").map(function(){    
      $(this).bootstrapSwitch();
    });

Resets the switch to default settings. And, I can figure out how to get it to inherit the PHP settings that came over via AJAX. And, I can't figure out any Jquery API against the $(this).bootstrapSwitch() to reconfigure the settings.

Using this approach from your post here https://github.com/kartik-v/yii2-tabs-x/issues/18...
 var opts = window[$('#switch-id').attr('data-krajee-bootstrapSwitch')];
              $("#switch-id").bootstrapSwitch(opts); // reinitialize switch input
doesn't work at all
neither did
 var opts = window[$('#switch-id').attr('data-kv-switch')];

I've been struggling with this for 24 hours. I'd appreciate any response you can offer.

There's just no explanation for reinitialize with settings. Or, Jquery settings of a new AJAX loaded switch

Thanks!

**Comment by kartik-v** [comment]
Simpler method is to render widget using pjax within a yii2 `Pjax::begin()` and `Pjax::end()` sections - then set the `SwitchInput::pjaxContainerId` within the switch input widget.

The Krajee plugins will auto initialize if you have set pjaxContainerId.

**Comment by newscloud** [comment]
Thanks so much for responding quickly. I very much appreciate it !!!!

I don't have any experience with Pjax and have had trouble in the past with the Gii generated grids that use Pjax.

As I started diving into your answer, I just didn't know all the places that pjax code has to be to make that work.

I ended up getting it work (I think):
$("input[name='place-chooser']").map(function(){
        //$(this).bootstrapSwitch();
        $(this).bootstrapSwitch('onText','<i class="glyphicon glyphicon-ok"></i>&nbsp;choose');
        $(this).bootstrapSwitch('offText','<i class="glyphicon glyphicon-remove"></i>');
        $(this).bootstrapSwitch('onColor','success');
        $(this).bootstrapSwitch('handleWidth',70);
        $(this).bootstrapSwitch('size','small');

And I'm needing to move a lot of my JavaScript architecture around to be preloaded and delegate events. But, I have proof of concept that seems to be working. :)
